internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/MetaSystemStoreTest.java

```diff
@@ -287,6 +289,42 @@ public void testThinClientMetaStoreBasedRepository() throws InterruptedException
     }
   }
 
+  @Test(timeOut = 120 * Time.MS_PER_SECOND)
+  public void testRequestBasedMetaStoreBasedRepository() throws InterruptedException {
+    String regularVeniceStoreName = Utils.getUniqueString("venice_store");
+    createStoreAndMaterializeMetaSystemStore(regularVeniceStoreName);
+    D2Client d2Client = null;
+    NativeMetadataRepository nativeMetadataRepository = null;
+    try {
+      d2Client = D2TestUtils.getAndStartD2Client(veniceLocalCluster.getZk().getAddress());
+      ClientConfig<StoreMetaValue> clientConfig =
+          getClientConfig(regularVeniceStoreName, d2Client).setUseRequestBasedMetaRepository(true);
+      // Not providing a CLIENT_META_SYSTEM_STORE_VERSION_MAP, should use the default value of 1 for system store
+      // current version.
+      VeniceProperties backendConfig = new PropertyBuilder().put(CLIENT_USE_SYSTEM_STORE_REPOSITORY, true)
+          .put(CLIENT_SYSTEM_STORE_REPOSITORY_REFRESH_INTERVAL_SECONDS, 1)
+          .build();
+      nativeMetadataRepository = NativeMetadataRepository.getInstance(clientConfig, backendConfig);
+      nativeMetadataRepository.start();
+      // ThinClientMetaStoreBasedRepository implementation should be used since CLIENT_USE_META_SYSTEM_STORE_REPOSITORY
+      // is set to true without enabling other feature flags.
+      Assert.assertTrue(nativeMetadataRepository instanceof RequestBasedMetaRepository);
```

Comment:
The comment doesn't match the assertion and also this is a weird place to put a  test for `RequestBasedMetaRepository` in a file called `MetaSystemStoreTest`. Can we add a new test file where we run a more comprehensive integration test with a DVC? i.e push some new versions or make some store config changes and make sure the DVC pick up those changes. You can see examples like the recently added `testBatchOnlyMaterializedViewDVCConsumer`. You probably don't need a `VeniceTwoLayerMultiRegionMultiClusterWrapper`, a single region will be sufficient.